src/main/java/com/autotune/analyzer/services/ListDeploymentsInNamespace.java

```diff
@@ -0,0 +1,87 @@
+/*******************************************************************************
+ * Copyright (c) 2022, 2022 Red Hat, IBM Corporation and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package com.autotune.analyzer.services;
+
+import com.autotune.common.target.kubernetes.service.KubernetesServices;
+import com.autotune.common.target.kubernetes.service.impl.KubernetesServicesImpl;
+import com.autotune.utils.AutotuneConstants;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.stream.Collectors;
+
+public class ListDeploymentsInNamespace extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) {
+        KubernetesServices kubernetesServices = null;
+        try {
+            String namespace = request.getParameter(AutotuneConstants.JSONKeys.NAMESPACE);
+            if (null == namespace) {
+                String inputData = request.getReader().lines().collect(Collectors.joining());
+                JSONObject inputJson = new JSONObject(inputData);
+                if (!inputJson.has(AutotuneConstants.JSONKeys.NAMESPACE)) {
+                    response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                    namespace = null;
+                } else {
+                    if (null == inputJson.getString(AutotuneConstants.JSONKeys.NAMESPACE)
+                            || inputJson.getString(AutotuneConstants.JSONKeys.NAMESPACE).isBlank()
+                            || inputJson.getString(AutotuneConstants.JSONKeys.NAMESPACE).isEmpty()) {
+                        response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                        namespace = null;
+                    } else {
+                        namespace = inputJson.getString(AutotuneConstants.JSONKeys.NAMESPACE);
+                    }
+                }
+            }
+            if (null != namespace) {
```

Comment:
this should just be an else statement?

Reply:
Ah! yes you are right, Not sure why I have written in that way. I don't currently remember the reason now, but to me it looks like I have written `null != namespace` case first and for some reason I need to add a null check which I placed on top of it so it turned out to be two if cases.

Corrected it now. Lemme know if any other changes needed.

Reply:
Ah! sorry I jus overlooked earlier.

Here is why i have added `null != namespace`
- First I'll check if there is a namespace param passed
- if not then i'll check if there is a json body which has key `namespace`
- if I find namespace i'll search deployments
- if not i'll return invalid namespace

So in this way this api is flexible to accept namespace as `autotuneUrl/query/listDeployments?namespace=<namespace>` or json sent to `autotuneUrl/query/listDeployments`